### PR TITLE
Add gumroad extension

### PR DESCRIPTION
**Pull request by eddiesigner**
## Description

View your sales and products from Gumroad.

## Screencast

gumroad-01
gumroad-02
gumroad-03

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/gumroad/src/view-sales.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+          icon={{ source: Icon.Coins, tintColor: Color.Magenta }}
+          accessories={[
+            {
+              text: sale.formatted_total_price,
+              icon: { source: Icon.BankNote, tintColor: Color.Green },
+            },
+          ]}
+          actions={
+            <ActionPanel>
+              <Action.Push title="Show Details" target={<SaleDetails sale={sale} />} />
```
[the remark this answers is not in the document]
Comment (eddiesigner) [reply extensions/gumroad/src/view-sales.tsx]: Added, thanks!

File: extensions/gumroad/src/product-details.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+export function ProductMetadata(props: { product: Product }) {
+  const item = props.product;
+
+  return (
+    <Detail.Metadata>
+      <Detail.Metadata.TagList title={"Price"}>
+        <Detail.Metadata.TagList.Item text={item.formatted_price} color={Color.Green} />
+      </Detail.Metadata.TagList>
+      <Detail.Metadata.Label title={"ID"} text={item.id} />
```
[the remark this answers is not in the document]
Comment (eddiesigner) [reply extensions/gumroad/src/product-details.tsx:46]: Thanks, now both fields will be shown as long as they are available

File: extensions/gumroad/src/view-sales.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    } else {
+      setPageUrl(`${BASE_URL}${SALES_ENDPOINT}?${TOKEN_PARAM}&product_id=${newValue}`);
+    }
+    setSales([]);
+    revalidate();
+  };
+
+  return (
+    <List
+      isLoading={isLoadingSales || sales.length === 0}
```
[the remark this answers is not in the document]
Comment (eddiesigner) [reply extensions/gumroad/src/view-sales.tsx]: Good catch, thanks! I already fixed the issue.

**Comment by eddiesigner** [comment]
Hi @xmok,
Thank you so much for your feedback! Your suggestions make sense to me so I have decided to make the relevant changes 👍🏼

**Comment by xmok** [comment]
Works great! 
We could make some changes especially in terms of native pagination and not having to manually "load more" but that can be handled in future updates.
For now, LGTM 🚀.

**Comment by xmok** [comment]
Oh great catch 😅 then!
LGTM 🚀

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Comment by eddiesigner** [comment]
@pernielsentikaer Awesome, thank you! 🚀

**Pull request merged**